### PR TITLE
9331 Make authentication pages accessible

### PR DESCRIPTION
**Pull request by johnatawnclementawn**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

Improve a user's ability to navigate through the authentication pages using a screen reader by including more landmarks and improving semantic HTML structure. This also includes CSS changes to use floating labels - a pattern that places labels where a place holder would normally be, then on input element focus, moves the label above the input.

### Issues Solved

#9331 

### Checklist

-   [ ] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by johnatawnclementawn** [comment]
@apeters @phudson-he

**Review by apeters — CHANGES_REQUESTED** [review]
This doesn't render correctly in RTL, could you please fix?

File: arches/app/media/css/arches.scss [hunk 1]
```diff
@@ -5386,144 +5386,7 @@ div.dropdown-menu.open {
     position: absolute;
 }
```
Comment (apeters) [inline arches/app/media/css/arches.scss:5388]: there are additional password css rules that should be moved over to the _authentication.scss file

```
.password-rules {
    margin-top: -5px;
    font-size: 12px;
    color: #555;
}

.password-rule {
    color: #888;
    font-size: 13px;
    padding-left: 5px;
    padding-bottom: 3px;
}

.password-rule span {
    padding-left: 3px;
    font-size: 12px;
}

.password-rule i {
    font-size: 11px;
}
```

Also can you make sure that these values render correctly in RTL. 

Screenshot 2023-04-14 at 5 59 51 PM

File: arches/app/media/css/pages/_authentication.scss [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+.floating-label-input:focus {
+    outline: 1;
+    box-shadow: 1;
+    background: transparent;    
+}
+
+.floating-label-input:focus ~ .floating-label, 
+.floating-label-input:valid ~ .floating-label {
+    top: -2em;
```
Comment (apeters) [inline arches/app/media/css/pages/_authentication.scss:52]: change this to `-2.5em`

File: arches/app/media/css/pages/_authentication.scss [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+.floating-label-input.change-password:focus ~ .floating-label, 
+.floating-label-input.change-password:valid ~ .floating-label {
+    top: -0.85em; 
+}
+
+.floating-label {
+    position: absolute;
+    padding: 0 0.5em 0 0.5em;
+    top: 1em;
```
Comment (apeters) [inline arches/app/media/css/pages/_authentication.scss:64]: change the padding to `padding: 0.75em;` 
remove the `top` setting

**Review by apeters — CHANGES_REQUESTED** [review]

File: arches/app/media/css/pages/_authentication.scss (hunk 3 above)
Comment (johnatawnclementawn) [reply arches/app/media/css/pages/_authentication.scss]: Without `top`, the labels in the login and new user pages are not placed properly. Was your suggestion specifically regarding the change password template?

Comment (apeters) [reply arches/app/media/css/pages/_authentication.scss]: @johnatawnclementawn I'm not seeing that.  If I make the floating label css this:
```
.floating-label {
    position: absolute;
    padding: 0.75em;
    pointer-events: none;
    user-select: none;
    transition: 0.2s ease all;
    -moz-transition: 0.2s ease all;
    -webkit-transition:0.2s ease all;
}
```
things seem to render well.  Can you confirm?

Comment (johnatawnclementawn) [reply arches/app/media/css/pages/_authentication.scss]: I can confirm, although the transitions seem less smooth - top: 0; fixes that. I can remove if necessary.

**Review by apeters — APPROVED** [review]
look good!

**Pull request merged**